ratpack-core/src/main/java/ratpack/server/internal/ServerRegistry.java

```diff
@@ -91,6 +95,11 @@ public static Registry buildBaseRegistry(RatpackServer ratpackServer, Imposition
         .maxContentLength(serverConfig.getMaxContentLength())
       );
 
+      DnsNameResolverBuilder resolverBuilder = new DnsNameResolverBuilder()
+        .eventLoop(execController.getEventLoopGroup().next())
+        .channelType(TransportDetector.getDatagramChannelImpl())
+        .socketChannelType(TransportDetector.getSocketChannelImpl());
```

Comment:
This is essentially the bare minimum for the resolver. 

Should we provide some type of configuration via `ServerConfig` to provide tunings here or just ask the user to provide their own implementation of `AddressResolverGroup` on the Registry?

Reply:
It will be very beneficial if we can configure the dns query timeout at the least as default. 
But, there is already a way to provide few configurations with this change:  https://github.com/netty/netty/commit/4f72cdf2333f682410e23e1eb5d77bc66b186cf4#diff-e7ae8540514a0e3caa415abc27aaee9fe93b7732d318d05a9534686447a162b9L146 .
Other tuning configs would become tricky to expose here just because they are internal to netty and are in plenty.

Reply:
I didn't see that previously @amityatagiri, good find.
It does look like the query time is configured based on the the `timeout:` option in `/etc/resolv.conf` (https://github.com/netty/netty/blob/4.1/resolver-dns/src/main/java/io/netty/resolver/dns/UnixResolverDnsServerAddressStreamProvider.java#L334).
So if this is configurable here, what about the benefit of providing a config option inside of Ratpack?
I'd probably lean toward not providing any options here since they are configurable and obtained from the OS by default and ask the user to supply a fully configured `AddressResolverGroup` if they want to tweak things.

Reply:
1.9.x branch is still pointing to netty `4.1.48.Final`. We may want to match it with the master branch or to the latest netty.

Reply:
Yes, 
`master` is the 2.x branch right now, that's why it's updated. But we can do that updated in `1.9.x` as well. @amityatagiri can you open another PR that is targeted to `1.9.x` which that dependency change?